Deu esse erro aqui, mas não tô entendendo o porque. Deixei o nome igual em todas as chamadas.

C:\Users\Gregory\Documents\GitHub\TrabEDA2017.1\App\Headers>gcc List.h
List.h:19:27: error: unknown type name 'DataNode'
 NodeList * createNodeList(DataNode *d);//cria um novo nÃ³
                           ^
In file included from File.h:16:0,
                 from Menu.h:16,
                 from Data.h:26,
                 from List.h:29:
Unissex.h: In function 'Insert':
Unissex.h:21:20: warning: implicit declaration of function 'createNodeList' [-Wi
mplicit-function-declaration]
             novo = createNodeList(Data);
                    ^
Unissex.h:21:18: warning: assignment makes pointer from integer without a cast [
-Wint-conversion]
             novo = createNodeList(Data);
                  ^
List.h: At top level:
List.h:47:12: error: conflicting types for 'createNodeList'
 NodeList * createNodeList(DataNode *d){
            ^
In file included from File.h:16:0,
                 from Menu.h:16,
                 from Data.h:26,
                 from List.h:29:
Unissex.h:21:20: note: previous implicit declaration of 'createNodeList' was her
e
             novo = createNodeList(Data);
                    ^

C:\Users\Gregory\Documents\GitHub\TrabEDA2017.1\App\Headers>

--- a/App/Headers/List.h
+++ b/App/Headers/List.h
@@ -16,19 +16,19 @@ typedef struct headList {
 
 
 List * createList(); //Cria, Inicializa e Retorna um Ponteiro para uma Lista
-NodeList * createNodeList(DataNode *d);//cria um novo nó
 int removeNodeList(List* L, int mat);//remove um nó pela matrícula
 NodeList * buscaMatriculaList ( int inputMatricula, List * root );//Função para Busca por Matricula em Lista
 int InsertList(List *L, NodeList *NewNode); //Função para inserir um nó na lista, ordenado pela matrícula
 NodeList * buscaNomeList(List *L, char *n, char *sn); //função busca um nó na lista pelo nome e sobrenome, e retorna um ponteiro pro nó
 NodeList * buscaEqualsList(int inputMatricula, List * root ); //Função que retorna se existe a matricula exata na lista
-void deleteNodeList(NodeList * input); //Remove Node de Lista
 void showAllList(List *L);
 
 //dependências
 #include "Data.h"
 #include <strings.h>
 
+NodeList * createNodeList(DataNode *d);//cria um novo nó
+void deleteNodeList(NodeList * input); //Remove Node de Lista
 
 //Função que Aloca um Espaço para criar a Lista e Retorna o Ponteiro da Lista
 List * createList()

--- a/App/Headers/List.h
+++ b/App/Headers/List.h
@@ -135,11 +135,11 @@ NodeList * buscaNomeList(List *L, char *n, char *sn){
 
     NodeList *aux = L->next;
 
-    while((strcasecmp(n, aux->data->nome) != 0) || (strcasecmp(sn, aux->data->sobrenome) != 0)  && aux->next != NULL){
+    while((strcasecmp(n, aux->data->nome) != 0) || ((strcasecmp(sn, aux->data->sobrenome) != 0)  && (aux->next != NULL)) ){
         aux = aux->next;
     }
 
-    if(aux->next == NULL && (strcasecmp(n, aux->data->nome) != 0) || (strcasecmp(sn, aux->data->sobrenome) != 0))
+    if( ((aux->next == NULL) && (strcasecmp(n, aux->data->nome) != 0)) || (strcasecmp(sn, aux->data->sobrenome) != 0))
         return NULL;
     else
         return aux;

--- a/App/Headers/Menu.h
+++ b/App/Headers/Menu.h
@@ -42,6 +42,7 @@ int mainExecute(int command, void * Tads[]) {
         case 5:
             return 0;
     }
+    return 0;
 }
 
 //Função que Valida os Dados que Recebe do Menu

--- a/App/Headers/Unissex.h
+++ b/App/Headers/Unissex.h
@@ -29,6 +29,7 @@ int Insert(void * Tads[], DataNode * Data, int type ) {
             // Tads[0] = InsertTree( (RootTree *) Tads[type], Data) @Função para Inserir na Arvore | Tratamento de Erro Interno
             // break;
     }
+    return 0;
 }
 
 //Cria alguma estrutura de Dado e retorna o seu ponteiro
@@ -39,6 +40,7 @@ void * Create(int type, DataNode * startData ) {
         //case 2:
             //return InsertTree( NULL, Data) @Função para Inserir na Arvore
     }
+    return NULL;
 }
 
 //Deleta algum nó pela Matricula

